### PR TITLE
Make selectable text mouse cursor configurable

### LINKED ISSUE #104595: Should `SelectionArea` cursor lose to click cursor?
When wrapping something selectable like `Text` in an `InkWell` with `onTap`, the cursor changes unexpectedly.

Should the `SelectionArea` cursor be made to "lose" to the click cursor?

```
import 'package:flutter/material.dart';

class SelectionAreaCursorRepro extends StatelessWidget {
  const SelectionAreaCursorRepro({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: SelectionArea(
        child: Scaffold(
          body: InkWell(
            onTap: () => {},
            child: const Padding(
              padding: EdgeInsets.all(32.0),
              child: Text('Hello'),
            ),
          ),
        ),
      ),
    );
  }
}
```

https://user-images.githubusercontent.com/6655696/170219503-68196aea-c1b0-429a-8b10-f316fcde3fdf.mp4

### PR DESCRIPTION
**Pull request by bleroux**
## Description

This PR introduces `DefaultSelectionStyle.mouseCursor` to configure the mouse cursor over selectable text.
It also applies this solution to `InkResponse` to make the mouse cursor win over the default one provided by selectable `Text` for many Material components (such as buttons). 

### Before

https://user-images.githubusercontent.com/840911/233627729-ddf98e2a-444d-4c6d-a6d5-f521982f48dd.mov

### After

https://user-images.githubusercontent.com/840911/233627718-8871a68f-d33c-44cf-b4a1-91bb1fcdf076.mov

## Related Issue

Fixes https://github.com/flutter/flutter/issues/104595

## Tests

Adds 6 tests.

### CODE REVIEW

**Comment by guidezpl** [comment]
Nice work! However there are quite a few components with clickable parts that include text. These include: 

- FAB / extended FAB
- Chips 
- Tabs
- Navigation (rail, drawer, bar)
- Top app bar
- XListTiles (e.g. `CheckboxListTile`)
- Menus
- Date and time pickers

Rather than handle these individually, can we add it a level below, say to `InkWell` and see which components still need to be handled individually?

**Comment by bleroux** [comment]
> Rather than handle these individually, can we add it a level below, say to InkWell and see which components still need to be handled individually?

I agree with this. Let me try to update this PR.
First goal of this PR was to get feedback for the `DefaultSelectionStyle.mouseCursor` solution I proposed. I choosed Material buttons to illustrate how this can be used but it makes sense to use it for InkWell to reduce code duplication.

File: packages/flutter/lib/src/widgets/default_selection_style.dart [hunk 1]
```diff
@@ -58,6 +82,12 @@ class DefaultSelectionStyle extends InheritedTheme {
   /// The background color of selected text.
   final Color? selectionColor;
 
+  /// The [MouseCursor] for mouse pointers that are hovering over selectable
+  /// Text widgets.
```
Comment (guidezpl) [inline packages/flutter/lib/src/widgets/default_selection_style.dart:86]:
```suggestion
  /// The [MouseCursor] for mouse pointers hovering over selectable [Text] widgets.
```

File: packages/flutter/lib/src/widgets/default_selection_style.dart [hunk 2]
```diff
@@ -41,8 +41,32 @@ class DefaultSelectionStyle extends InheritedTheme {
   const DefaultSelectionStyle.fallback({ super.key })
     : cursorColor = null,
       selectionColor = null,
+      mouseCursor = null,
       super(child: const _NullWidget());
 
+  /// Creates a default selection style that overrides the selection styles in
+  /// scope at this point in the widget tree.
```
Comment (guidezpl) [inline packages/flutter/lib/src/widgets/default_selection_style.dart:48]: nit: https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

**Review by guidezpl — APPROVED** [review]
Nice, please update PR description.

```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

**Comment by reza00farjam** [comment]
Thanks for fix. On which channel and which version this fix is available?

**Comment by bleroux** [comment]
This fix is available on the master channel.

**Comment by guidezpl** [comment]
pro tip: https://github.com/flutter/flutter/commit/d4f884e000194d050a745ccd2b09029225b89636 indicates the branches the merge commit is present on

**Comment by LucaIaconelli** [comment]
Thanks for the fix. Is it also available in stable or only on master?

**Comment by guidezpl** [comment]
As long as stable isn't listed on that commit, it's not available. Since beta releases are monthly, expect it in the next one.

**Comment by LucaIaconelli** [comment]
@guidezpl still not available on stable?

**Comment by federicopolesello** [comment]
@LucaIaconelli I see that is present in stable from version `3.13.0`

https://docs.flutter.dev/release/release-notes/release-notes-3.13.0

**Comment by reza00farjam** [comment]
For future reference (both for myself and others), here's an example of how this output can be utilized effectively:  

```dart
SelectionArea(
  child: DefaultSelectionStyle.merge(
    mouseCursor: SystemMouseCursors.click,
    child: Text('Clickable and selectable text'),
  ),
);
```

**Pull request merged**